### PR TITLE
Create expectation with autogenerated flag set to true

### PR DESCRIPTION
**Pull request by anthonyburdi**
## What

Create an expectation with the autogenerated flag set to true. Since the autogenerated field is not a GX Core concept, we add it to the expectation config before sending to mercury.

Note that this PR skips the existing integration test. We will have a followup PR to re-enable. For now, here is a demo of this functionality working e2e with both the agent and the backend: https://gxlabs.zoom.us/clips/share/A2F3MRZZZU5JUEpSWFRKLXN2X0NtMExrazZBAQ

- App changes: https://github.com/greatexpectationslabs/mercury/pull/4114
- REST API changes: https://github.com/greatexpectationslabs/mercury/pull/4118
- Agent changes: This PR: https://github.com/great-expectations/cloud/pull/583

## Why

We need to determine whether an expectation is autogenerated or not so that we can handle differently in the UI.

### CODE REVIEW

File: tests/integration/actions/test_generate_schema_change_expectations.py [hunk 1]
```diff
@@ -66,29 +104,12 @@ def token_env_var_local():
...
-    local_mercury_db_datasource: PostgresDatasource,
-) -> TableAsset:
-    data_asset_name = "local-mercury-db-organizations-table"
-    data_asset = local_mercury_db_datasource.get_asset(name=data_asset_name)
-    yield data_asset
-
-
+@pytest.mark.skip(
+    "This test fails due to a connection error when attempting to call the backend. It is a problem with the test setup, not the code which was demoed, and this test passes locally."
+)
```
Comment (anthonyburdi) [inline tests/integration/actions/test_generate_schema_change_expectations.py:109]: Test is marked skipped until we can figure out the issues running on the CI infrastructure.

File: great_expectations_cloud/agent/models.py [hunk 2]
```diff
@@ -109,6 +109,7 @@ class GenerateSchemaChangeExpectationsEvent(EventBase):
     )
     datasource_name: str
     data_assets: Sequence[str]
+    # TODO: Remove data_asset_to_expectation_suite_name from the event
```
Comment (Shinnnyshinshin) [inline great_expectations_cloud/agent/models.py:112]: Is this a ticket? I'm in the one that added this parameter, so I'd be happy to do the removal work too :)

Comment (anthonyburdi) [reply great_expectations_cloud/agent/models.py:112]: It is! I forgot to mark it "TODO": https://greatexpectations.atlassian.net/browse/ZELDA-1166

File: tests/integration/actions/test_generate_schema_change_expectations.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply tests/integration/actions/test_generate_schema_change_expectations.py:109]: Sounds good. Really appreciate the work that you've done with this.

File: tests/integration/actions/test_generate_schema_change_expectations.py [hunk 3]
```diff
@@ -43,12 +42,51 @@ def seed_and_cleanup_test_data(context: CloudDataContext):
...
+        conn.execute(sa.text(query))
+        conn.commit()
+
     # Yield
     yield table_data_asset, suite
 
     # clean up
-    data_source.delete_asset(name="local-mercury-db-checkpoints-table")
+
+    # Mark the validation as not gx_managed
```
Comment (Shinnnyshinshin) [inline tests/integration/actions/test_generate_schema_change_expectations.py:75]: :rocket:

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM!

File: tests/integration/actions/test_generate_schema_change_expectations.py (hunk 1 above)
Comment (anthonyburdi) [reply tests/integration/actions/test_generate_schema_change_expectations.py:109]: Here is the ticket to re-enable: https://greatexpectations.atlassian.net/browse/ZELDA-1175

**Pull request merged**